BUILD.md

```diff
@@ -1,66 +1,138 @@
-alias waf="$PWD/modules/waf/waf-light"
-```
+## Basic usage ##
 
-You can also define the alias or create a function in your shell rc file (e.g.
-`~/.bashrc`).
+There are several commands in the build system for advanced usages, but here we
+list some basic and more used commands as example.
 
-You can read the [Waf Book](https://waf.io/book/) if you want to learn more
-about Waf.
+* **Build ArduCopter**
 
-## Calling waf ##
+    Here we use minlure as an example of Linux board. See more advanced usages
+    below to get a list of supported boards.
 
-Waf should always be called from the ardupilot's root directory.
+    ```sh
+    ./waf configure --board minlure
+    ./waf copter
+    ```
 
-Differently from the make-based build, with Waf there's a configure step
-to choose the board to be used (default is `sitl`):
+    The first command should be called only once or when you want to change a
+    configuration option. One configuration often used is the `--board` option to
+    switch from one board to another one. For example we could switch Pixhawk and
+    build again:
 
-```bash
-# Configure the Linux board
-waf configure --board=linux
-```
+    ```sh
+    ./waf configure --board px4-v2
+    ./waf copter
+    ```
 
-Waf build system is composed of commands. For example, the above command
-(`configure`) is for configuring the build. Consequently, in order to build, a
-"build" command is issued, thus `waf build`. That is the default command, so
-calling just `waf` is enough:
+* **List available boards**
 
-```bash
-# Build programs from bin group
-waf
 
-# Waf also accepts '-j' option to parallelize the build.
-waf -j8
-```
+    It's possible to get a list of supported boards on ArduPilot with the command
+    below
 
-To clean things up, use the `clean` or `distclean` command:
+    ```sh
+    ./waf list_boards
 
-```bash
-# Clean the build products, but keep configure information
-waf clean
+    ```
+
+* **Clean the build**
+
+    Commands `clean` and `distclean` can be used to clean the objects produced by
+    the build. The first keeps the configure information while the second cleans
+    everything.
+
+    Cleaning the build is very often not necessary and discouraged. We do
+    incremental builds reducing the build time by orders of magnitude.
+
+
+* **Upload or install**
+
+    Pixhawk board has and upload option so it uploads to the firmware built to
+    a connected board. In order to use it, pass the option `--upload` to build
+    command:
+
+    ```sh
+    ./waf --targets bin/arducopter-quad --upload
+    ```
+
+    Currently Linux boards don't support the upload option, but there's a install
+    command, which will install to a certain directory. This can be used by
+    distributors to create .deb, .rpm or other package types:
+
+    ```sh
+    ./waf copter
+    DESTDIR=/my/temporary/location ./waf install
+    ```
+
+* **Use different targets**
+
+    The build commands in the items above use `copter` as the target. This
```

Comment:
No need to know about "group"... That's very specific to waf and is in the advanced section. "Target" was used because it's what people are used in Makefile world and as "what will be produced by the command".

Example: git show

I don't want to see in the help of git explaining how that is implemented, if the binary execs itself with arguments, etc. For the user it's just a target/subcommand/action/argument.